src/Native/FactorizationMachineNative/FactorizationMachineCore.cpp

```diff
@@ -105,9 +105,9 @@ EXPORT_API(void) CalculateIntermediateVariablesNative(int fieldCount, int latent
     *response = linearResponse + latentResponse;
 }
 
-// Calculate the stochastic gradient and update the model. 
+// Calculate the stochastic gradient and update the model.
 // The /*const*/ comment on the parameters of the function means that their values should not get altered by this function.
-EXPORT_API(void) CalculateGradientAndUpdateNative(float lambdaLinear, float lambdaLatent, float learningRate, int fieldCount, int latentDim, float weight, int count,
```

Comment:
Given the difficulty of reading AVX code, please add a comment:
// This function implements Algorithm 2 in [https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf](https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf)
.

Please add the same line to SSE and FMA counterparts.